### PR TITLE
1 tx burning

### LINKED ISSUE #1417: 1 function 1 transaction
## Description

Don't have functions that could create multiple transactions internally

## Motivation

Better error handling, clearer logic

## Requirements

1. Change claiming and consolidation to only send one transaction at a time, where the user has to provide the output ids (return error if they don't fit in a single tx)
2. Change minting to return an error if there is no existing alias output, add a new method instead to create an alias output
3. Change burn destroy methods to return an error if there is still an output owned by the output we're about to burn/destroy? That way the users would need to burn/destroy the outputs at the end of the ownership chain first (this should also make it harder to burn/destroy something by accident)

## Open questions (optional)

Any strong reasons to not do this? For claiming and consolidation it might require the users to create a loop to consolidate/claim all outputs, but that shouldn't be hard to do

## Are you planning to do it yourself in a pull request?

Yes/No.

### PR DESCRIPTION
**Pull request by Thoralf-M**
# Description of change

1 tx burning

## Links to any relevant issues

Fixes #1417 

## Type of change

- Breaking change (fix or feature that would cause existing functionality to not work as expected)

## How the change has been tested

Describe the tests that you ran to verify your changes.

Make sure to provide instructions for the maintainer as well as any relevant configurations.

## Change checklist

Tick the boxes that are relevant to your changes, and delete any items that are not.

- [ ] I have followed the contribution guidelines for this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have checked that new and existing unit tests pass locally with my changes

### CODE REVIEW

File: src/account/operations/transaction/input_selection.rs [hunk 1]
```diff
@@ -53,9 +53,9 @@ impl AccountHandle {
             }
 
             let selected_transaction_data = try_select_inputs(
+                Vec::new(),
                 custom_inputs,
```
Comment (Thoralf-M) [inline src/account/operations/transaction/input_selection.rs]:
```suggestion
                custom_inputs,
                Vec::new(),
```

File: src/account/operations/helpers/time.rs [hunk 2]
```diff
@@ -16,18 +19,29 @@ pub(crate) fn can_output_be_unlocked_now(
...
 
-        let unlock_address = unlock_conditions.locked_address(output_address, current_time);
-        // The address that can unlock the output needs to belong to the account
-        if !account_addresses.iter().any(|a| a.address.inner == *unlock_address) {
-            return false;
-        };
+            let unlock_address = unlock_conditions.locked_address(address, current_time);
+            // The address that can unlock the output needs to belong to the account
+            if !account_addresses.iter().any(|a| a.address.inner == *unlock_address)
+                && !alias_and_nft_addresses.iter().any(|a| a == unlock_address)
```
Comment (thibault-martinez) [inline src/account/operations/helpers/time.rs]: Use `contains` instead?

File: src/account/operations/transaction/high_level/burning_melting/destroy_alias.rs [hunk 3]
```diff
@@ -9,24 +9,46 @@ use iota_client::block::{
...
-        self.sweep_address_outputs(Address::Alias(AliasAddress::new(alias_id)), &address)
-            .await?;
+        let current_time = self.client().get_time_checked().await?;
+
+        let mut owned_outputs = Vec::new();
+
+        for output_data in self.list_unspent_outputs(None).await? {
+            if can_output_be_unlocked_now(
+                // Don't provide any addresses here, since we're only interested in outputs that can be unlocked by
+                // the nft address
```
Comment (thibault-martinez) [inline src/account/operations/transaction/high_level/burning_melting/destroy_alias.rs]:
```suggestion
                // the alias address
```

File: src/account/operations/helpers/time.rs [hunk 4]
```diff
@@ -16,18 +19,29 @@ pub(crate) fn can_output_be_unlocked_now(
...
+            let address = address_unlock_condition.address();
 
-        let unlock_address = unlock_conditions.locked_address(output_address, current_time);
-        // The address that can unlock the output needs to belong to the account
-        if !account_addresses.iter().any(|a| a.address.inner == *unlock_address) {
-            return false;
-        };
+            let unlock_address = unlock_conditions.locked_address(address, current_time);
+            // The address that can unlock the output needs to belong to the account
+            if !account_addresses.iter().any(|a| a.address.inner == *unlock_address)
```
Comment (thibault-martinez) [inline src/account/operations/helpers/time.rs:27]: contains as well :trollface: ?

Comment (Thoralf-M) [reply src/account/operations/helpers/time.rs:27]: Doesn't work because we check for .address.inner, no?

Comment (thibault-martinez) [reply src/account/operations/helpers/time.rs:27]: Right, overlooked it

**Review by thibault-martinez — APPROVED** [review]

**Pull request merged**